### DISCUSSION TITLE
Bug: gridcell ARIA role incorrectly added to Day button element

### DISCUSSION THREAD

**Discussion by rihardsgravis** [gpbl/react-day-picker#1674] [Support]
### Bug description

The ARIA `gridcell` role is added to the Day button, but that is an incorrect role for the button since it does not represent a cell in a grid – it is in fact a child of a cell. Accessibility tests using the component are failing.

### Expected behaviour

The role for the button could be `button` or omitted at all since it's a HTM  button component and the role for it is clear.

**Comment by gpbl** [comment]
@rihardsgravis the role is correct, in v8.5 we changed it to be a `gridcell` and the parent `td` element to be ignored in the accessibility tree:

<img width="800" alt="Screenshot 2023-02-02 at 10 11 19" src="https://user-images.githubusercontent.com/120693/216363171-551d4f9d-49a8-4380-9b5b-6a8952065ace.png">

I'm sorry this change broke your tests: it should be easy to update them - targeting the `gridcell`  in the grid.

For example, to get the day button you'd need to update:

```diff
- screen.getByRole('button', {
+ screen.getByRole('gridcell', {
-  name: format(day, 'do MMMM (EEEE)')
+  name: day.getDate().toString()
});
```

I'm thinking if there's a way to apply this change without having a mayor release...

**Reply by rihardsgravis** [reply]
Oh, sorry for not being clear on the tests failing – our tests are not failing because of a broken selector, but our [axe automated accessibility tests](https://github.com/dequelabs/axe-core) fail with the reason that  the `gridcell` role is not allowed for a `button` element. The reasoning for that is described [here](https://dequeuniversity.com/rules/axe/4.6/aria-allowed-role?application=AxeChrome)

**Reply by gpbl** [reply]
@rihardsgravis good to know, thanks for the heads up! Let me see if I can find a fix. Meanwhile, avoid upgrading to v8.5 until then.

**Comment by gpbl** [comment]
Main discussion here: https://github.com/gpbl/react-day-picker/discussions/1716
